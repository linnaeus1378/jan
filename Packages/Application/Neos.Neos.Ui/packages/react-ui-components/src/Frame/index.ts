import {themr} from '@friendsofreactjs/react-css-themr';

import identifiers from '../identifiers';
import Frame from './frame';

export default themr(identifiers.frame)(Frame);
