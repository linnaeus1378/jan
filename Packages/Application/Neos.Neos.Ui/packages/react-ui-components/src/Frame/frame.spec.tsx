import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import Frame, {FrameProps} from './frame';

describe('<Frame/>', () => {
    const props: FrameProps = {
        children: 'Foo children',
        contentDidUpdate: jest.fn(),
        mountTarget: 'foo',
        onLoad: jest.fn(),
        onUnload: jest.fn(),
        src: 'localhost',
    };

    it('should render correctly.', () => {
        const wrapper = shallow(<Frame {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<Frame {...props} className="fooClassName"/>);

        expect(wrapper.prop('className')).toContain('fooClassName');
    });

    it('should allow the propagation of additional props to the wrapper.', () => {
        const wrapper = shallow(<Frame {...props} frameBorder="0"/>);

        expect(wrapper.prop('frameBorder')).toBe('0');
    });
});
