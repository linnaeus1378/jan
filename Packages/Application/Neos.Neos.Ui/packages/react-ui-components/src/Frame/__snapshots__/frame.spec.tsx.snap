// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Frame/> should render correctly. 1`] = `
<iframe
  onLoad={[Function]}
/>
`;
